### ISSUE TITLE
How can I use parenthesis around already existing functions.

### ISSUE THREAD

**Issue by Saikedo** [daniel3735928559/guppy#133]
Imagine this, the user already have something like 3+5*2 and the user wants something like (3+5)*2 instead. How can I do this? It seems like the only support for parenthesis is to generate the () object.

**Comment by daniel3735928559** [comment]
Right now, if you have 3+5*2 and select the 3+5 and press the `(` key while the 3+5 is selected, it will wrap only that part of the expression in parens, resulting in (3+5)*2 as desired. 

Or perhaps you knew that already and wanted something better, in which case I'm interested to know what kind of interface you had in mind?

**Comment by Saikedo** [comment]
@daniel3735928559 I am aware of the highlighting approach that you mentioned. I was wondering how complicated it would be to let the user put opening parenthesis wherever they want. Guppy will then automatically put a ghost parenthesis(basically a closing parenthesis with a bit different shading which acts as a placeholder to not break the underlying structure) at the end of the scope for the open parenthesis that the user puts. After this, the user would be allowed to put the actual closing parenthesis anywhere in the scope between the open parenthesis that they put and the placeholder close parenthesis that the library generated.

Does this explanation make sense?

**Comment by saivan** [comment]
I think this would be a very useful interface as well. Although you'd need to support implicit closing brackets. They'd get replaced as soon as the actual closing bracket gets added on.

**Comment by JMSS-Unknown** [comment]
+1 This is implemented in MathQuill and is very useful. For example if you type `1+2+3` then place an opening bracket in front of the one, a light bracket will appear after the 3. If you then place a closing bracket after the 2, the light bracket will disappear leaving with `(1+2)+3`

**Comment by JMSS-Unknown** [comment]
I have attempted to replicate MathQuill's behaviour with implicit parenthesis in #149. Please tell me if anything is missing @Saikedo and @saivan.

**Comment by saivan** [comment]
@JMSS-Unknown Wow that is awesome! I’ll jump onto it and have a look soon 👍

**Comment by saivan** [comment]
@daniel3735928559 would you consider making an org for guppy and adding some of us to it? It would be good to keep this project alive

**Comment by daniel3735928559** [comment]
Agreed #149 looks excellent! Apologies I have been busy, and have spent more of my free time working on applications using guppy than on the editor itself. 

The org is an excellent idea; I've now made: https://github.com/guppy-js. Let me spend some time this weekend merging in my local changes, reviewing the outstanding PRs and then migrating over to the org and adding people.

Thanks all for your help!

**Comment by saivan** [comment]
No worries, I'm excited for that :D Thanks @daniel3735928559
